test/ess-test-inf.el

```diff
@@ -89,16 +89,17 @@
   ;; (ess-async-command "{cat(1:5);Sys.sleep(5);cat(2:6)}\n" nil (get-process "R")
   ;;                    (lambda (proc) (message "done"))
   (with-r-running nil
-    (let (semaphore)
+    (let ((inf-proc *proc*)
```

Comment:
I now realise a cleaner syntax would explicitly bind the inferior variables like this:

```lisp
(with-r-running (inf-buf inf-proc) file ...)
```

The binding would be optional:


```lisp
(with-r-running (inf-buf) file ...)
(with-r-running nil file ...)
```